adapters/avail/src/verifier.rs

```diff
@@ -0,0 +1,62 @@
+use serde::{Deserialize, Serialize};
+use sov_rollup_interface::da::{BlockHeaderTrait, DaSpec, DaVerifier};
+use sov_rollup_interface::zk::ValidityCondition;
+use thiserror::Error;
+
+use crate::spec::DaLayerSpec;
+
+#[derive(Error, Debug)]
+pub enum ValidityConditionError {
+    #[error("conditions for validity can only be combined if the blocks are consecutive")]
+    BlocksNotConsecutive,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+/// A validity condition expressing that a chain of DA layer blocks is contiguous and canonical
+pub struct ChainValidityCondition {
+    pub prev_hash: [u8; 32],
+    pub block_hash: [u8; 32],
+}
+
+impl ValidityCondition for ChainValidityCondition {
+    type Error = ValidityConditionError;
+
+    fn combine<SimpleHasher>(&self, rhs: Self) -> Result<Self, Self::Error> {
+        if self.block_hash != rhs.prev_hash {
+            return Err(ValidityConditionError::BlocksNotConsecutive);
+        }
+
+        Ok(rhs)
+    }
+}
+
+pub struct Verifier;
+
+impl DaVerifier for Verifier {
+    type Spec = DaLayerSpec;
+
+    type Error = ValidityConditionError;
+
+    type ValidityCondition = ChainValidityCondition;
+
+    // Verify that the given list of blob transactions is complete and correct.
+    // NOTE: Function return unit since application client already verifies application data.
+    fn verify_relevant_tx_list<SimpleHasher>(
+        &self,
+        block_header: &<Self::Spec as DaSpec>::BlockHeader,
+        _txs: &[<Self::Spec as DaSpec>::BlobTransaction],
+        _inclusion_proof: <Self::Spec as DaSpec>::InclusionMultiProof,
+        _completeness_proof: <Self::Spec as DaSpec>::CompletenessProof,
+    ) -> Result<Self::ValidityCondition, Self::Error> {
+        let validity_condition = ChainValidityCondition {
+            prev_hash: *block_header.prev_hash().inner(),
+            block_hash: *block_header.hash().inner(),
+        };
+
+        Ok(validity_condition)
+    }
```

Comment:
@vibhurajeev, this implementation breaks soundness.

The only way for a rollup to work without a smart contract is to allow the end-user (light client) to verify that the rollup's fork choice rule was followed. If the verifier can't check this, then a malicious prover can create a *valid* proof which extends a non-canonical chain, ultimately causing honest light clients to accept the wrong rollup state. 

In this implementation, the SNARK doesn't commit to the transactions that it's processed. It commits to the block header, but since this function doesn't tie the actual bytes of data that the rollup processes back to the block header, that doesn't constrain the prover. Given a block header, the prover can pretend that the block contained whatever transactions he wants and the SNARK won't stop him. 

As we discussed offline, it's ok to push the work of verifying the transaction list outside of the SNARK - that's exactly what `ValidityConditions` were added for. But, the SNARK still has to commit to the actual set of blobs that it received from the prover, not just the alleged block header.

I think the correct way to implement this is roughly...

1. Add an additional field to the `ChainValidityCondition` for `tx_root`, and put the hash of the `_txs: &[<Self::Spec as DaSpec>::BlobTransaction]` argument in that field
2. Add a `check`er for the validity condition which downloads the data from a light client and enforces that it matches those nodes. 
3. Change the `combine` rule for validity conditions to ensure that when you check the validity condition for block N, you're also forced ot recursively check back to Genesis. (This is necessary since just because the prover gave the correct tx list in block N doesn't mean he gave the correct one in block N-1).